config/initializers/devise.rb

```diff
@@ -234,10 +234,11 @@ def i18n_message(default = nil)
   # Range for password length.
   config.password_length = 10..128
 
-  # Email regex used to validate email formats. It simply asserts that
-  # one (and only one) @ exists in the given string. This is mainly
+  # Email regex used to validate email formats. It asserts that
+  # one (and only one) @ exists in the given string and that the domain only
+  # contains numbers, letters and full stops. This is mainly
   # to give user feedback and not to assert the e-mail validity.
-  config.email_regexp = /\A[^@\s]+@[^@\s]+\z/
+  config.email_regexp = /\A[^@\s]+@[A-Za-z0-9.]+\z/
```

Comment:
I think I'd lean toward not maintaining our own, or subtly deviating from the Devise original. There is a built-in expression we could go for. 
```ruby
URI::MailTo::EMAIL_REGEXP.match? 'me@example,com'
# => false
```